compiler/hash-ast/src/parse.rs

```diff
@@ -178,22 +197,49 @@ where
                     );
                     let entry_resolver = ParModuleResolver::new(ctx, entry_module_ctx, scope);
 
+                    // @@Cleanup: we need to insert the contents of interactive into the module builder...
+                    //            At the moment, this isn't the cleanest way of going about the problem, we're
+                    //            overwriting the previous content of the interactive session with the current
+                    //            input. This is incorrect because we want to preserve the lines of interactive
+                    //            that are considered to be valid (this is a much deeper problem actually.)
```

Comment:
Why would we want to preserve the previous lines?

Reply:
This comment is probably not well formed. It's not so much that we should store previous lines, but rather the generated AST from previous statements needs to be stored somewhere. Or it might not be the AST that is stored, but rather the typchecked results.

Reply:
I imagine we need to keep track of TypecheckCtx, but not much else.